### PR TITLE
refactor: feature toggle using hathor unleash client

### PR DESCRIPTION
**Pull request by andreabadesso**
### Acceptance Criteria
- We should use the hathor unleash client instead of the official unleash client

### Security Checklist
- [X] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

**Review by alexruzenhack — APPROVED** [review]

File: src/sagas/featureToggle.js [hunk 1]
```diff
@@ -102,86 +102,37 @@ export function* monitorFeatureFlags(currentRetry = 0) {
...
     // Spawn so it's detached from the current thread
     yield spawn(handleInitFailed, currentRetry);
-  } finally {
-    if (yield cancelled()) {
-      yield call(() => unleashClient.stop());
-    }
-  }
-}
-
-export function* setupUnleashListeners(unleashClient) {
```
Comment (pedroferreira1) [inline src/sagas/featureToggle.js:153]: How are you handling this now?

File: src/sagas/featureToggle.js [hunk 2]
```diff
@@ -102,86 +102,37 @@ export function* monitorFeatureFlags(currentRetry = 0) {
...
-    const featureToggles = mapFeatureToggles(unleashClient.toggles);
+    // At this point, unleashClient.fetchToggles() already fetched the toggles
+    // (this will throw if it hasn't)
+    const featureToggles = mapFeatureToggles(unleashClient.getToggles());
 
     yield put(setFeatureToggles(featureToggles));
     yield put(featureToggleInitialized());
   } catch (e) {
-    console.error('Error initializing unleash');
-    unleashClient.stop();
```
Comment (pedroferreira1) [inline src/sagas/featureToggle.js:137]: Why did you remove this?

File: src/sagas/featureToggle.js [hunk 3]
```diff
@@ -102,86 +102,37 @@ export function* monitorFeatureFlags(currentRetry = 0) {
...
-    yield call(() => unleashClient.updateContext(options));
     yield put(setUnleashClient(unleashClient));
 
-    // Listeners should be set before unleashClient.start so we don't miss
-    // updates
-    yield fork(setupUnleashListeners, unleashClient);
-
-    // Start without awaiting it so we can listen for the
-    // READY event
-    unleashClient.start();
```
Comment (pedroferreira1) [inline src/sagas/featureToggle.js:115]: You don't need to start anymore?

Comment (andreabadesso) [reply src/sagas/featureToggle.js:115]: No, we are doing the loop in the saga

File: src/sagas/featureToggle.js (hunk 1 above)
Comment (andreabadesso) [reply src/sagas/featureToggle.js:153]: In the `fetchToggleRoutine` saga, it checks if the toggles were updated and yields a `FEATURE_TOGGLE_UPDATE` action

**Review by pedroferreira1 — APPROVED** [review]
